src/tests/ftest/util/pool_security_test_base.py

```diff
@@ -104,30 +97,50 @@ def verify_daos_pool_cont_result(self, result, action, expect, err_code):
             action (str): daos pool read or write.
             expect (str): expecting pass or deny.
             err_code (str): expecting deny of RC code.
-
-        Return:
-            none.
-
         """
-        if expect.lower() == 'pass':
-            if result.exit_status != 0 or result.stderr_text != "":
-                self.fail(
-                    "##Test Fail on verify_daos_pool {}, expected Pass, but "
-                    "Failed.".format(action))
+        if isinstance(result, dict):
```

Comment:
To reduce code reuse, how about replacing the two checks with methods that handle the different `result` types:
```
    @staticmethod
    def _command_failed(self, result):

        if isinstance(result, dict):
            return result["status"] != 0 or result["error"] is not None
        else:
            return result.exit_status != 0 or result.stderr_text != ""

    @staticmethod
    def _command_missing_err_code(result, err_code):

        if isinstance(result, dict):
            return err_code not in result["error"]
        else:
            return (err_code not in result.stderr_text and
                    err_code not in result.stdout_text)
```